src/transformers/modeling_tf_pytorch_utils.py

```diff
@@ -54,6 +56,14 @@ def convert_tf_weight_name_to_pt_weight_name(tf_name, start_prefix_to_remove="",
         - transpose: `TransposeType` member indicating whether and how TF2.0 and PyTorch weights matrices should be
           transposed with regards to each other
     """
+    if name_scope:
+        if not tf_name.startswith(name_scope):
+            raise ValueError(
+                f"Weight name {tf_name} does not start with name_scope {name_scope}. This is an internal error "
+                "in Transformers, so (unless you were doing something really evil) please open an issue to report it!"
+            )
+        tf_name = tf_name[len(name_scope) :]  # We can't use remove_prefix because we need to support older Python
```

Comment:
Not sure the comment brings much to the table (except obnoxiously pointing out our knowledge of Python 3.9+ functions ;-) )

Reply:
It's only slightly obnoxious! (I'll remove it)